src/c20_server/job_queue.py

```diff
@@ -3,21 +3,29 @@
 implementation of Job_Queue class
 
 """
+import pickle
+import threading
 from c20_server import job_queue_errors
 
 
 class JobQueue:
 
-    def __init__(self):
-        self.unassigned_jobs_list = []
+    def __init__(self, database):
+        self.r_database = database
+        self.lock = threading.RLock
```

Comment:
After a long discussion with Prof. Coleman about using lock, we both agreed to have lock in job queue. But I decided to comment it out for now, since we did not implement the real Redis server yet.